insights-notification-schemas-java/src/main/resources/schemas/Action.json

```diff
@@ -0,0 +1,123 @@
+      "description": "Extra information common to all events sent.",
+      "title": "Context"
+    },
+    "events": {
+      "type": "array",
+      "items": {
+        "title": "Events",
+        "type": "object",
+        "properties": {
+          "metadata": {
+            "type": "object",
+            "properties": {},
+            "title": "Event's metadata",
+            "description": "Placeholder for future details we might need"
+          },
+          "payload": {
+            "type": "object",
+            "properties": {},
+            "description": "Your application payload. All the data required by the app to compose the various messages (Email, webhook, etc) after transformation",
+            "title": "Event payload"
+          }
+        },
+        "additionalProperties": false,
+        "required": [
+          "payload"
+        ]
+      }
+    },
+    "recipients": {
+      "type": "array",
+      "default": [],
+      "items": {
+        "type": "object",
+        "properties": {
+          "only_admins": {
+            "type": "boolean",
+            "default": false,
+            "title": "Only admins",
+            "description": "Setting to true sends an email to the administrators of the account. Setting to false sends an email to all users of the account. Note that this does not ignore the user preferences."
+          },
+          "ignore_user_preferences": {
+            "type": "boolean",
+            "default": false,
+            "title": "Ignore user preferences",
+            "description": "Setting to true ignores all the user preferences on this Recipient setting (It doesn't affect other configuration that an Administrator sets on their Notification settings). Setting to false (default) honors the user preferences."
+          },
+          "users": {
+            "type": "array",
+            "default": [],
+            "items": {
+              "type": "string"
+            },
+            "title": "Users",
+            "description": "List of users to direct the notification to. Note that this does not ignore the user preferences."
+          },
+          "groups": {
+            "type": "array",
+            "default": [],
+            "items": {
+              "type": "string"
+            },
+            "title": "Groups",
+            "description": "List of group ids to direct the notification to. Note that this does not ignore the user preferences."
+          }
+        },
+        "additionalProperties": false
+      },
+      "description": "Applications can add extra email recipients by adding entries to this array. This setting extends whatever the Administrators configured in their Notifications settings. This won't override notification's administrator settings - this will extend the settings."
+    }
+  },
+  "additionalProperties": false,
+  "required": [
+    "bundle",
+    "application",
+    "event_type",
+    "timestamp",
+    "account_id",
+    "events"
+  ]
```

Comment:
I suppose there's no way to add a `required` field to each declared property instead of having this section.

Reply:
> I suppose there's no way to add a required field to each declared property instead of having this section.

Yes, that's the json-schema way of specifing required fields

![this-is-the-way](https://c.tenor.com/0iCG-EWpDzMAAAAC/mandalorian-this-is-the-way.gif)